src/plugins/dashboard/public/application/listing/create_button.tsx

```diff
@@ -14,7 +14,7 @@ import {
 import type { DashboardProvider } from '../../types';
 
 interface CreateButtonProps {
-  dashboardProviders?: DashboardProvider[];
+  dashboardProviders?: { [key: string]: DashboardProvider } | null;
```

Comment:
why `| null`? If it's optional, I'd likely refactor so that we use `undefined` instead, which would already be accounted for.